DataFormats/BTauReco/interface/ShallowTagInfo.h

```diff
@@ -0,0 +1,39 @@
+#ifndef DataFormats_BTauReco_ShallowTagInfo_h
+#define DataFormats_BTauReco_ShallowTagInfo_h
+
+#include "DataFormats/BTauReco/interface/RefMacros.h"
+#include "DataFormats/BTauReco/interface/BaseTagInfo.h"
+#include "DataFormats/BTauReco/interface/CandIPTagInfo.h"
+#include "DataFormats/BTauReco/interface/CandSecondaryVertexTagInfo.h"
+#include "DataFormats/BTauReco/interface/TaggingVariable.h"
+
+namespace reco {
+
+class ShallowTagInfo : public BaseTagInfo {
+public:
+  ShallowTagInfo(void) { }
+
+  ShallowTagInfo(
+    const TaggingVariableList & list,
+    const edm::RefToBase<Jet> & jetref) :
+      list_(list),
+      jetRef_(jetref) { }
+
+  virtual ~ShallowTagInfo(void) { }
+
+  virtual ShallowTagInfo* clone(void) const { return new ShallowTagInfo(*this); }
+
+  virtual edm::RefToBase<Jet> jet(void) const { return jetRef_; }
+
+  virtual TaggingVariableList taggingVariables(void) const { return list_; }
+
+protected:
+  /*const*/ TaggingVariableList  list_;
```

Comment:
[this comment goes to 90X as well]

isn't this plain TaggingVariableList as a data member a problem for (evolving) persistency?
Other types like TemplatedSecondaryVertexTagInfo derive the list on the fly based on the current library values of ```enum TaggingVariableName ``` to provide consistent meaning.
Here, if one decides to save ShallowTagInfo, it will become non-obviously meaningless after the TaggingVariableName enum changes.

It may be worth to denote this transient or redo the storage logic.